myndla-api/src/main/scala/no/ndla/myndlaapi/db/migrationwithdependencies/V16__MigrateResourcePaths.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Part of NDLA myndla-api
+ * Copyright (C) 2024 NDLA
+ *
+ * See LICENSE
+ */
+
+package no.ndla.myndlaapi.db.migrationwithdependencies
+
+import no.ndla.myndlaapi.integration.TaxonomyApiClient
+import no.ndla.network.NdlaClient
+import org.flywaydb.core.api.migration.{BaseJavaMigration, Context}
+import scalikejdbc.{DB, DBSession, scalikejdbcSQLInterpolationImplicitDef}
+
+import java.util.UUID
+
+trait V16__MigrateResourcePaths {
+  this: TaxonomyApiClient & NdlaClient =>
+
+  class V16__MigrateResourcePaths extends BaseJavaMigration {
+    private val chunkSize = 100;
+
+    override def migrate(context: Context): Unit = DB(context.getConnection)
+      .autoClose(false)
+      .withinTx { implicit session =>
+        migrateResources
+      }
+
+    private def migrateResources(implicit session: DBSession): Unit = {
+      val count        = countResources.get
+      var numPagesLeft = (count / chunkSize) + 1
+      var offset       = 0L
+
+      while (numPagesLeft > 0) {
+        allResources(offset * chunkSize).foreach { case (id, resourceType, path) =>
+          resourceType match {
+            case "article" | "learningpath" | "multidisciplinary" | "topic" =>
+              val updatedPath = taxonomyApiClient.resolveUrl(path)
+              updatedPath.map(updateResource(UUID.fromString(id), _))
+            case _ => ()
+          }
+        }
+        numPagesLeft -= 1
+        offset += 1
+      }
+    }
+
+    private def updateResource(id: UUID, path: String)(implicit session: DBSession): Int = {
+      sql"update resources set path=$path where id = $id".update()
+    }
+
+    private def allResources(offset: Long)(implicit session: DBSession): Seq[(String, String, String)] = {
+      sql"select id, resource_type, path from resources order by id limit $chunkSize offset $offset"
+        .map(rs => {
+          (rs.string("id"), rs.string("resource_type"), rs.string("path"))
+        })
+        .list()
+    }
+
+    private def countResources(implicit session: DBSession): Option[Long] = {
+      sql"select count(*) from resources"
+        .map(rs => rs.long("count"))
+        .single()
+    }
```

Comment:
Kunne brukt `TableMigration` helperen jeg lagde for en liten stund siden for å abstrahere vekk rad-iterasjonen:

```suggestion
    override val tableName: String                  = "resources"
    override val whereClause: scalikejdbc.SQLSyntax = sqls""
    override def extractRowData(rs: WrappedResultSet): ResourceRow = ResourceRow(
      UUID.fromString(rs.string("id")),
      rs.string("resource_type"),
      rs.string("path")
    )

    override def updateRow(rowData: ResourceRow)(implicit session: DBSession): Int = {
      rowData.resourceType match {
        case "article" | "learningpath" | "multidisciplinary" | "topic" =>
          taxonomyApiClient
            .resolveUrl(rowData.path)
            .map { path => sql"update resources set path=$path where id = ${rowData.id}".update() }
            .get
        case _ => 0
      }
    }
```

Reply:
Og evt hivd inn chunksize støtte i den om det er lurere med 100 fremfor 1000 som det er i den klassen.